src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -93,63 +112,881 @@ static ColliderQuadInit D_809AFFD8 = {
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->unk_258 == 0) {
+        EnAm_SetupSleep(this);
+    }
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.world.rot.y;
+}
+
+void EnAm_MoveToHome(EnAm* this, GlobalContext* globalCtx) {
+    s16 yawToHome = Math_Vec3f_Yaw(&this->dyna.actor.world.pos, &this->dyna.actor.home.pos);
+
+    if (this->skelAnime.curFrame == 8.0f) {
+        this->dyna.actor.velocity.y = 12.0f;
+        this->dyna.actor.speedXZ = 6.0f;
+    } else if (this->skelAnime.curFrame > 11.0f) {
+        if (!(this->dyna.actor.bgCheckFlags & 1)) {
+            this->skelAnime.curFrame = 11;
+        } else {
+            Math_SmoothStepToS(&this->dyna.actor.world.rot.y, yawToHome, 1, 0xBB8, 0);
+
+            if (this->dyna.actor.bgCheckFlags & 2) {
+                this->unk_258--;
+            }
+
+            this->dyna.actor.velocity.y = 0.0f;
+            this->dyna.actor.speedXZ = 0.0f;
+            this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+            EnAm_SpawnEffects(this, globalCtx);
+
+            if (Actor_WorldDistXYZToPoint(&this->dyna.actor, &this->dyna.actor.home.pos) < 80.0f) {
+                EnAm_SetupRotateToInit(this);
+            }
+        }
+    }
+
+    // turn away from a wall if touching one
+    if ((this->dyna.actor.speedXZ != 0.0f) && (this->dyna.actor.bgCheckFlags & 8)) {
+        this->dyna.actor.world.rot.y = this->dyna.actor.wallYaw;
+        Actor_MoveForward(&this->dyna.actor);
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.world.rot.y;
+}
+
+void EnAm_RecoilFromDamage(EnAm* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.speedXZ < 0.0f) {
+        this->dyna.actor.speedXZ += 0.5f;
+    }
+
+    if ((this->dyna.actor.velocity.y <= 0.0f) && !EnAm_CanMove(this, globalCtx, -8.0f, this->dyna.actor.world.rot.y)) {
+        this->dyna.actor.speedXZ = 0.0f;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnAm_SetupLunge(this);
+        this->deathTimer = 64;
+        this->panicSpinRot = 0;
+    }
+}
+
+/**
+ * After doing 3 lunges, wait for 2 seconds before attacking again.
+ * Turn toward the player before lunging again.
+ */
+void EnAm_Cooldown(EnAm* this, GlobalContext* globalCtx) {
+    s16 yawDiff = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.world.rot.y;
+
+    if (yawDiff < 0) {
+        yawDiff = -yawDiff;
+    }
+
+    if (this->cooldownTimer != 0) {
+        this->cooldownTimer--;
+    } else {
+        if (this->skelAnime.curFrame == 8.0f) {
+            Math_SmoothStepToS(&this->dyna.actor.world.rot.y, this->dyna.actor.yawTowardsPlayer, 1, 0x1F40, 0);
+            this->dyna.actor.velocity.y = 12.0f;
+        } else if (this->skelAnime.curFrame > 11.0f) {
+            if (!(this->dyna.actor.bgCheckFlags & 1)) {
+                this->skelAnime.curFrame = 11;
+            } else {
+                if (yawDiff < 3500) {
```

Comment:
Angle in hex?

Reply:
its a clean number in decimal and we arent enforcing one over the other